.github/workflows/ci.yaml

```diff
@@ -87,3 +87,23 @@ jobs:
 
     - name: docs
       run: cargo doc --no-deps
+
+  check_features:
+    name: Check all feature combinations works properly
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+
+    - name: Install stable
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        override: true
+
+    - name: Install cargo-hack
+      run: cargo install cargo-hack
+
+    - name: Check all feature combinations
+      # * `--feature-powerset` - run for the feature powerset of the package
+      # * `--no-dev-deps` - build without dev-dependencies to avoid https://github.com/rust-lang/cargo/issues/4866
+      run: cargo hack check --feature-powerset --no-dev-deps
```

Comment:
If this check takes a long time, it can limit the max number of simultaneous feature flags by passing the `--depth` flag. (like [futures-rs](https://github.com/rust-lang/futures-rs/blob/59f62b04d92a236a9909e3ef0ed6cf973ad9d593/.github/workflows/ci.yml#L218), [hyper](https://github.com/hyperium/hyper/blob/42587059e6175735b1a8656c5ddbff0edb19294c/.github/workflows/CI.yml#L115), etc. do) AFAIK `--depth 2` should be sufficient in most cases as described in https://github.com/taiki-e/cargo-hack/issues/58.